### ISSUE TITLE
az network application-gateway rule create fails for redirect-rule as of cli version 2.45

### ISSUE THREAD

**Issue by SanderMeesterIndepender** [Azure/azure-cli#25408]
> ### `az feedback` auto-generates most of the information requested below, as of CLI version 2.0.62

**Related command**
```
az network application-gateway rule create 
      --gateway-name my-appgateway 
      --resource-group rg-my-appgateway 
      --name 'my-http-redirect-rule' 
      --http-listener 'my-http-listener' 
      --rule-type Basic 
      --redirect-config my-redirect-config 
      --priority 333
```

**Describe the bug**
command fails on azure cli 2.45 (it works on azure cli 2.43) with the following error:
`ERROR: Multiple backend address pools found. Specify --address-pool explicitly.`

**To Reproduce**
1. create http-listener on port 443, complete with backend pool and rule
2. create http-listener on port 80
3. create redirect-config as described below
4. create rule as described above

**Expected behavior**
redirect-rule should bind redirect-config and http-listener 

**Environment summary**
azure devops Task         : Azure CLI Version      : 2.216.0

azure-cli                         2.45.0
core                              2.45.0
telemetry                          1.0.8

Dependencies:
msal                              1.20.0
azure-mgmt-resource             21.1.0b1

Python (Windows) 3.10.8 (tags/v3.10.8:aaaf517, Oct 11 2022, 16:37:59) [MSC v.1933 32 bit (Intel)]

**Additional context**

my-redirect-config looks like this:

```
      az network application-gateway redirect-config create 
      --gateway-name my-appgateway 
      --resource-group rg-my-appgateway 
      --name my-redirect-config 
      --type Permanent 
      --target-listener my-https-listener 
      --include-path true 
      --include-query-string true
```

**Comment by yonzhan** [comment]
@necusjz for awareness

**Comment by necusjz** [comment]
@SanderMeesterIndepender
> create http-listener on port 443, complete with backend pool and rule

Which backend pool is using for http-listener?

There is a default backend pool created when we use `az network application-gateway create`, which named `appGatewayBackendPool`.
> ERROR: Multiple backend address pools found. Specify --address-pool explicitly.

It means current application gateway has multiple backend pools (except default). When use `--address-pool` to specify it, can this problem be solved?

**Comment by SanderMeesterIndepender** [comment]
/unresolve

Thanks for taking the time with me to solve this. The command works fine with cli 2.43, but it fails on cli 2.45, so is suspect something has changed somewhere. But anyway, here is my complete script. So yes; the address-pool is specified for the http-listener on 443.
```

      az network application-gateway address-pool create `
      --gateway-name my-appgateway `
      --resource-group rg-my-appgateway `
      --name 'my-app.azurewebsites.net' `
      --servers 'my-app.azurewebsites.net'

      $sslCertificateResourceId = $(az network application-gateway ssl-cert list --resource-group rg-my-appgateway --gateway-name my-appgateway --query "[?name=='my-certificate'].[id]" --output tsv)
      az network application-gateway http-listener create `
      --gateway-name my-appgateway `
      --resource-group rg-my-appgateway `
      --name 'my-app-https-listener' `
      --frontend-port port_443 `
      --frontend-ip appGwPrivateFrontendIp `
      --host-name 'my-hostname.example.com' `
      --ssl-cert "$sslCertificateResourceId"

      az network application-gateway probe create `
      --resource-group rg-my-appgateway `
      --gateway-name my-appgateway `
      --name 'my-app-probe' `
      --path "/" `
      --protocol https `
      --host 'my-hostname.example.com'
      
      az network application-gateway http-settings create `
      --resource-group rg-my-appgateway `
      --gateway-name my-appgateway `
      --name 'my-app-http-settings' `
      --protocol 'Https' `
      --port 443 `
      --timeout 30 `
      --enable-probe true `
      --probe 'my-app-probe' `
      --cookie-based-affinity 'Enabled' `
      --affinity-cookie-name 'ApplicationGatewayAffinity' `
      --connection-draining-timeout 60

      az network application-gateway rule create `
      --gateway-name my-appgateway `
      --resource-group rg-my-appgateway `
      --name 'my-app-default-rule' `
      --http-listener 'my-app-https-listener' `
      --address-pool 'my-app.azurewebsites.net' `
      --http-settings 'my-app-http-settings' `
      --rule-type Basic `
      --rewrite-rule-set 'default-independer-rewrite-rule-set' `
      --priority 42

      ##################
      # http => https redirection
      ##################     

      az network application-gateway http-listener create `
      --gateway-name my-appgateway `
      --resource-group rg-my-appgateway `
      --name 'my-app-http-listener' `
      --frontend-port port_80 `
      --frontend-ip appGwPrivateFrontendIp `
      --host-name 'my-hostname.example.com' 

      az network application-gateway redirect-config create `
      --gateway-name my-appgateway `
      --resource-group rg-my-appgateway `
      --name my-app-redirect-config `
      --type Permanent `
      --target-listener my-app-https-listener `
      --include-path true `
      --include-query-string true

      az network application-gateway rule create `
      --gateway-name my-appgateway `
      --resource-group rg-my-appgateway `
      --name 'my-app-http-redirect-rule' `
      --http-listener 'my-app-http-listener' `
      --rule-type Basic `
      --redirect-config my-app-redirect-config `
      --priority 43      
```

**Comment by necusjz** [comment]
@SanderMeesterIndepender Thanks for your detailed procedures! I knew what happened and will fix it in the next release.

BTW, is there any workaround so far? Like explicitly add `--address-pool` in the last command.

**Comment by SanderMeesterIndepender** [comment]
There does not seem to be a workaround.

If I add `--address-pool` to the last command (which is weird, but ok), I'm greeted with a new error:
```
      az network application-gateway rule create `
      --gateway-name my-appgateway `
      --resource-group rg-my-appgateway `
      --name 'my-app-http-redirect-rule' `
      --http-listener 'my-app-http-listener' `
      --rule-type Basic `
      --redirect-config my-app-redirect-config `
      --address-pool 'my-app.azurewebsites.net' `
      --priority 43      
```
results in: `Multiple backend settings found. Specify --http-settings explicitly.`

If I then add `--http-settings` to the last command, I'm greeted with a new error (which makes sense):
```
      az network application-gateway rule create `
      --gateway-name my-appgateway `
      --resource-group rg-my-appgateway `
      --name 'my-app-http-redirect-rule' `
      --http-listener 'my-app-http-listener' `
      --rule-type Basic `
      --redirect-config my-app-redirect-config `
      --address-pool 'my-app.azurewebsites.net' `
      --http-settings 'my-app-http-settings' `
      --priority 43      
```
results in `Message: Request Routing Rule /<snip>/my-app-http-redirect-rule with RedirectConfiguration cannot reference a BackendAddressPool.`

So for now it seems that rolling back the cli is my only option.

**Comment by trichling** [comment]
Hi,

I am facing the same issue. Our deployment pipelines that provision clusters stopped working as Azure Pipelines Hosted Agents picked up az Version 2.45.0.

I can confirm the findings of @SanderMeesterIndepender: it asks for backend pool, if provided it asks for http-listener than for http-settings and then complains that you can not specify redirect-config and backend address pool together (which makes sense).

Is there a timeframe as of when this will be fixed? Bacuase I could not find a way to downgrade the CLI version inside of a hosted agent.

**Comment by necusjz** [comment]
@trichling I'll compose a fix PR in Feb and it'll be fixed in 2.46.0 on around 03/07.

**Comment by trichling** [comment]
> @trichling I'll compose a fix PR in Feb and it'll be fixed in 2.46.0 on around 03/07.

Thanks for the quick response!

**Comment by marlowp** [comment]
@trichling - we also ran into the same issue and the following command successfully reverted the version of the CLI for us on a ubuntu-20.04 Azure DevOps hosted agent:

```yaml
- task: PowerShell@2
  displayName: 'Revert to az cli version 2.44.0'
  inputs:
    targetType: 'inline'
    pwsh: true
    script: |
      pip install -Iv azure-cli==2.44.0 --extra-index-url https://azurecliprod.blob.core.windows.net/edge
```
